src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/SymbolTable.cs

```diff
@@ -1579,64 +1572,38 @@ private MethodSymbol AddMethodToSymbolTable(MemberInfo member, AggregateSymbol c
                 methodSymbol.SetConvNext(callingAggregate.GetFirstUDConversion());
                 callingAggregate.SetFirstUDConversion(methodSymbol);
             }
+
             ACCESS access;
-            if (method != null)
+            if (member.IsPublic)
             {
-                if (method.IsPublic)
-                {
-                    access = ACCESS.ACC_PUBLIC;
-                }
-                else if (method.IsPrivate)
-                {
-                    access = ACCESS.ACC_PRIVATE;
-                }
-                else if (method.IsFamily)
-                {
-                    access = ACCESS.ACC_PROTECTED;
-                }
-                else if (method.IsAssembly || method.IsFamilyAndAssembly)
-                {
-                    access = ACCESS.ACC_INTERNAL;
-                }
-                else
-                {
-                    Debug.Assert(method.IsFamilyOrAssembly);
-                    access = ACCESS.ACC_INTERNALPROTECTED;
-                }
+                access = ACCESS.ACC_PUBLIC;
+            }
+            else if (member.IsPrivate)
+            {
+                access = ACCESS.ACC_PRIVATE;
+            }
+            else if (member.IsFamily)
+            {
+                access = ACCESS.ACC_PROTECTED;
+            }
+            else if (member.IsFamilyOrAssembly)
```

Comment:
This also changes the order in testing for `IsFamilyOrAssembly` and if false inferring `IsAssembly || IsFamilyAndAssembly` rather than the other way around. Why test two properties when you can test one?